packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -312,6 +331,16 @@ class SemanticsNode extends AbstractNode {
     }
   }
 
+  Set<SemanticsTag> _tags = new Set<SemanticsTag>();
+
+  /// Tag the [SemanticsNode] with [tag].
```

Comment:
this should talk about how if you plan to add a tag conditionally, then you need to carefully think about how and when to call markNeedsSemantics so that it has the right effect. Also, when to call reset() vs when to call removeTag.